### DISCUSSION TITLE
access to library 'age' is not allowed for database user

### DISCUSSION THREAD

**Discussion by I33Buckler** [apache/age#1197] [Q&A]
Have successfully built the apache/age extension from source for Postgresql 15. 
Have ensured that the database user I have created has access to the schema ag_catalog.
Also have created the link from /usr/lib/postgresql/15/lib/age.so to /usr/lib/postgresql/15/lib/plugins/age.so.
running the statement "load 'age'" as user postgres works. However as the db user it results in the message "ERROR:  access to library "age" is not allowed".
Any assistance welcome.

**Comment by vladiksun** [comment]
For non-super user we always use
```
LOAD 'age';
SET search_path = ag_catalog, "$user", public;
```

We add those automatically for all scripts (java service)

**Comment by I33Buckler** [comment]
Turns out there was a corrupted age extension in the database cluster which was unable to be removed. 
I have rebuilt the cluster and installed a fresh extension. No further problem.
